bin/utest/vm_map.c

```diff
@@ -0,0 +1,73 @@
+#include "utest.h"
+
+#include <assert.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+int test_sharing_memory_simple(void) {
+  size_t pgsz = getpagesize();
+  char *map =
+    mmap(NULL, pgsz, PROT_READ | PROT_WRITE, MAP_ANON | MAP_SHARED, -1, 0);
+  assert(map != (char *)MAP_FAILED);
+
+  pid_t pid = fork();
+  assert(pid >= 0);
+
+  if (pid == 0) {
+    /* child */
+    strcpy(map, "Hello, World!");
+    exit(0);
+  }
+
+  /* parent */
+  int status;
+  assert(waitpid(-1, &status, 0) == pid);
+  assert(WIFEXITED(status));
```

Comment:
You can use the `wait_for_child_exit()` helper function from `util.h`.

Reply:
I didn't know this function. Many thanks! I will change it! :smiley: